File: adventofcode-2024/src/main/java/nl/mvdr/adventofcode/adventofcode2024/day24/Part1.java
package nl.mvdr.adventofcode.adventofcode2024.day24;

import java.util.stream.Stream;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.LongSolver;

public class Part1 implements LongSolver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part1.class);

    @Override
    public long solve(Stream<String> lines) {
        return Device.parse(lines.toList()).resolveZ();
    }
    
    public static void main(String[] args) {
        var instance = new Part1();

        var result = instance.solve("input-day24-2024.txt");

        LOGGER.info(result);
    }
}
 

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day01/Part1.kt
package nl.mvdr.adventofcode.adventofcode2025.day01

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.point.Direction
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart1(lines: Sequence<String>): Int {
    var dial = 50
    var zeroes = 0
    for (line in lines) {
        dial = rotate(dial, line)
        if (dial == 0) {
            zeroes++
        }
    }

    return zeroes
}

private fun rotate(dial: Int, instruction: String): Int {
    val direction = Direction.parse(instruction.take(1))
    val distance = instruction.substring(1).toInt()

    val directionMultiplier = when(direction) {
        Direction.LEFT -> -1
        Direction.RIGHT -> 1
        else -> throw IllegalArgumentException("Unexpected direction: {direction}")
    }

    return (dial + distance * directionMultiplier) % 100
}

fun main() {
    val result = FunctionSolver(::solvePart1).solve("input-day01.txt")
    logger.info { result }
}
